src/main/java/eu/donals/UserInterface.java

```diff
@@ -0,0 +1,261 @@
+
+        try {
+            for (String string : textBlock) {
+                if (string.length() > 98){ throw new DisplayError("text input string is too long: " + string); }
+                if (counter > 30){ throw new DisplayError("too many text input strings to display: " + textBlock); }
+
+                tg.putString(31, counter, clearLine);
+                tg.putString(31, counter, string);
+
+                counter++;
+            }
+
+            screen.refresh();
+        } catch (DisplayError e) {
+            e.printStackTrace();
+            System.exit(1);
+        }
+    }
+
+    // displays text within the text box
+    // @string  must be no more than 98 chars long, if it is longer, stringSplit() will attempt to make it fit
+    public void displayText(String string) throws IOException {
+        String clearLine = "                                                                                                  ";
+        List<String> splitString = new ArrayList<String>();
+
+        try {
+            if (string.length() > 98){
+                splitString = stringSplit(string);
+                if (splitString != null){
+                    displayText(splitString);
+                    return;
+                } else { throw new DisplayError("text input string is too long: " + string); }
+            }
+
+            // clears the text box
+            for (int i = 22; i < 30; i++) {
+                tg.putString(31, i, clearLine);
+            }
+
+            tg.putString(31, 22, string);
+            screen.refresh();
+        } catch (DisplayError e) {
+            e.printStackTrace();
+            System.exit(1);
+        }
+    }
+
+    // wraps readUserInput() and supplies the correct information so that input can be taken from the input box
+    public String takeInput() throws IOException {
+        String clearLine = "                                                                                                  ";
+        TerminalPosition termPos = new TerminalPosition(31, 35);
+
+        try {
+            return readUserInput(termPos);
+        } finally {
+            tg.putString(31, 35, clearLine);
+            screen.setCursorPosition(termPos);
+        }
+    }
+
+    // @initialPromptPosition: the position of the cursor before any input is read
+    private String readUserInput(TerminalPosition initialPromptPosition) throws IOException {
+        String userInput = "";
+        int posX = initialPromptPosition.getColumn();
+        int posY = initialPromptPosition.getRow();
+        KeyStroke keyStroke = new KeyStroke(' ', false, false);
+
+        while (keyStroke.getKeyType() != KeyType.Enter) {
+            keyStroke = screen.readInput();
+
+            switch (keyStroke.getKeyType()) {
+                case Character:
+                    if (userInput.length() > 20) {
```

Comment:
These loops could be split up in several functions. There should not be more than two layers per loop.